spec/mailers/activity_mailer_spec.rb

```diff
@@ -1,18 +1,20 @@
 require 'rails_helper'
 
 RSpec.describe ActivityMailer, type: :mailer do
-  let(:template_path) { File.join(__dir__, '../fixtures/activity_mailer/test_email') }
   let(:user) { create(:user, name: 'tester') }
-  let(:mail) { ActivityMailer.email(user, user, template_path) }
+  let(:notification) { create(:user_notification) }
+  let(:mail) { ActivityMailer.email(user, user, notification) }
   let(:text) { mail.body.parts.find { |part| part.content_type.start_with?('text/plain') }.to_s }
   let(:html) { mail.body.parts.find { |part| part.content_type.start_with?('text/html') }.to_s }
 
   it 'sets the correct headers' do
-    expect(mail.subject).to eq('test mailer')
+    pending 'an actual notification template'
```

Comment:
I've tried before. Now the mailer requires a notification not a path. We must stub the helper method . But it not thread-safe right? You said it before.